esmvalcore/_config/_config.py

```diff
@@ -54,7 +55,19 @@ def _load_extra_facets(project, extra_facets_dir):
 def get_extra_facets(project, dataset, mip, short_name, extra_facets_dir):
     """Read configuration files with additional variable information."""
     project_details = _load_extra_facets(project, extra_facets_dir)
-    return project_details.get(dataset, {}).get(mip, {}).get(short_name, {})
+
+    def fnfilter(patterns, name):
+        """Return the subset of the list `patterns` that match `name`."""
+        return [pat for pat in patterns if fnmatch.fnmatchcase(name, pat)]
+
+    extra_facets = {}
+    for dataset_ in fnfilter(project_details, dataset):
+        for mip_ in fnfilter(project_details[dataset_], mip):
+            for var in fnfilter(project_details[dataset_], short_name):
+                facets = project_details[dataset_][mip_][var]
+                extra_facets.update(facets)
+
+    return extra_facets
```

Comment:
If I understand correctly, this adds the ability to use Unix-style filename expansion for keys in the extra facets.

First, I am a bit uneasy about applying something that is clearly meant for filenames and path to something else. I think at a minimum we should have a comment here, stating clearly that, yes this is for filenames, but we use it for something else, and that works fine and is awesome.

Second, I think this extension of the syntax should be documented.

Third, this is sufficiently different from moving the institutes that I would consider doing this in a separate PR. At a minimum, it requires a change to the PR title, IMHO.

Reply:
> If I understand correctly, this adds the ability to use Unix-style filename expansion for keys in the extra facets.

Yes

> First, I am a bit uneasy about applying something that is clearly meant for filenames and path to something else. I think at a minimum we should have a comment here, stating clearly that, yes this is for filenames, but we use it for something else, and that works fine and is awesome.

I also considered using a regular expression, but I felt that the wildcards used for file selection would be more familiar to most users. For example, we also allow these for selecting ancestors in the recipe and tasks from the command line with the `--diagnostics` flag.

> Second, I think this extension of the syntax should be documented.

Good point, I'll work on improving it

> Third, this is sufficiently different from moving the institutes that I would consider doing this in a separate PR. At a minimum, it requires a change to the PR title, IMHO.

I really need a feature like this here, to be able to specify that the institute should be added to any mip and any variable. I'll update the pull request title and description.

Reply:
> > First, I am a bit uneasy about applying something that is clearly meant for filenames and path to something else. I think at a minimum we should have a comment here, stating clearly that, yes this is for filenames, but we use it for something else, and that works fine and is awesome.
> I also considered using a regular expression, but I felt that the wildcards used for file selection would be more familiar to most users. For example, we also allow these for selecting ancestors in the recipe and tasks from the command line with the `--diagnostics` flag.

I understand completely. In fact, I was rather surprised that the `fnmatch` docs are so filesystem-centric since it seems such a useful feature elsewhere. That's part of what made me weary. But I'll accept that, would just like it to be a bit more explicit for the benefit of a future maintainer/reader, i.e. rename `fnfilter` to something that doesn't imply `f`ile`n`ame and reword the one-line docstring to mention this. I, for example, needed a few minutes to convince myself that this code was not written with filenames in mind.

> > Second, I think this extension of the syntax should be documented.
> Good point, I'll work on improving it
:+1: 

> > Third, this is sufficiently different from moving the institutes that I would consider doing this in a separate PR. At a minimum, it requires a change to the PR title, IMHO.
> I really need a feature like this here, to be able to specify that the institute should be added to any mip and any variable. I'll update the pull request title and description.

If it was me, I'd probably split it up, get the extended syntax in first, and follow immediately with the institute moving. But it's ok.